src/traces/histogram/defaults.js

```diff
@@ -51,8 +50,9 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
 
     var hasAggregationData = traceOut[aggLetter];
     if(hasAggregationData) coerce('histfunc');
+    coerce('histnorm');
 
-    handleBinDefaults(traceIn, traceOut, coerce, [sampleLetter]);
+    // Note: bin defaults are now handled in Histogram.cleanData
```

Comment:
`_module.cleanData` [*is* in `supplyDefaults`](https://github.com/plotly/plotly.js/pull/3044/files#diff-ad4f76ccd6044ed16514297078e13b84R451) so this doesn't cause problems. I'll add a test though to 🔒 it.

Given its location in the pipeline - and the uses we're accreting for it - `cleanData` is probably not a great name, perhaps we should change it to `_module.crossTraceDefaults`?

Reply:
> _module.cleanData is in supplyDefaults so this doesn't cause problems. 

Oh right, I got confused with the only [`cleanData`](https://github.com/plotly/plotly.js/blob/67a05c707e02aed77674e7a0c557a56907338421/src/plot_api/helpers.js#L206-L390). Great!

> is probably not a great name, perhaps we should change it to `_module.crossTraceDefaults`?

This would be great! No need to do in this PR of course. A new issue or a comment in https://github.com/plotly/plotly.js/issues/749 would be good enough for now.

Reply:
:lock: autobin/bins in `validate` -> https://github.com/plotly/plotly.js/pull/3044/commits/76eee4b992eacbebff1f00ee025c7696eeb7f5d3

`cleanData` -> `crossTraceDefaults` -> https://github.com/plotly/plotly.js/pull/3044/commits/5d930cee2d0df51ad6bbb99a0df17a4eb2fe64dc